skbase/testing/utils/deep_equals.py

```diff
@@ -105,6 +119,23 @@ def _is_pandas(x):
         return False
 
 
+def _is_npndarray(x):
+
+    clstr = type(x).__name__
+    if clstr in ["ndarray"]:
+        return True
+
+
+def _coerce_list(x):
+    """Coerce x to list."""
+    if not isinstance(x, (list, tuple)):
+        x = [x]
+    if isinstance(x, tuple):
+        x = list(x)
+
+    return x
+
+
 def _pandas_equals(x, y, return_msg=False):
```

Comment:
Doesn't pandas have testing functions to assert objects are equal? Is there a reason we aren't leveraging those?

Reply:
not added in this PR, but originally added to allow for dealing with nested `pandas`. `pandas` onboard checkers don't (or didn't) play along well with the case where you nest `pandas` in `pandas`, since `!=` on `pandas` results in not a boolean but an iterable.

The case where you nest `pandas` in `pandas` occurs for the `nested_univ` mtype, the "original" mtype used for panel tasks in `sktime` (but now discouraged since `pandas` is acting up on newer versions when it comes to nesting itself).

Reply:
generally, `deep_equals` supports equality checking where you arbitrarily nest the supported types, including `pandas`. Given the funny composite types in time series, I think it is fair to have it in `sktime`. In `skbase`, don't know.

Reply:
Yea -- I considered asking the same question about the _fh comparisons. In the end we can't be all things to all packages. And it may not make sense to have time series specific items, but for now we can leave them. It would be good if `sktime` deep_equals could build on the `skbase` one (so we could remove the time series specific parts from `skbase`). Lets try to keep that in mind in the `sktime` PR that uses the `skbase` functionality.